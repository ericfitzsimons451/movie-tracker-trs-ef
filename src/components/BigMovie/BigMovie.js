import React from "react";
import "./BigMovie.scss";
import PropTypes from "prop-types";

export const BigMovie = ( {movie} ) => {
  console.log(movie)
  return (
    <div className='big-movie-card'>
    <div className='movie-poster'>
        <img src={`http://image.tmdb.org/t/p/original/${movie.poster_path}`} alt="poster" />
    </div>
    <div className='movie-details'>
        <h3 className="movie-title">{movie.name}</h3>
        <h2 className='release-date'>Released: {movie.release_date}</h2>
        <h2 className='vote-avg'>Rating: {movie.vote_average}</h2>
        <p className='overview'>{movie.overview}</p>
        {/* <button onClick={this.validateUser}>{btn}</button> */}
    </div>
</div>
)};

export default BigMovie;
