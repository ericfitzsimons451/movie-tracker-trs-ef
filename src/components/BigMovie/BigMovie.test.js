import React from 'react'
import { shallow } from 'enzyme'
import BigMovie from '../BigMovie/BigMovie'

describe('BigMovie', () => {
    let wrapper;
    it('should render the snapshot', () => {
        const mockMovie = {
            name: "Master Z: Ip Man Legacy",
            id: 450001,
            poster_path: "/2WfjB6FUDTIBVI2y02UGbnHR82s.jpg",
            release_date: "2018-12-20",
            vote_average: 5.5,
            overview:
            "After being defeated by Ip Man, Cheung Tin Chi is attempting to keep a low profile. While going about his business, he gets into a fight with a foreigner by the name of Davidson, who is a big boss behind the bar district. Tin Chi fights hard with Wing Chun and earns respect."
          }
        wrapper = shallow(<BigMovie movie={mockMovie} />)
        expect(wrapper).toMatchSnapshot()
    })
})