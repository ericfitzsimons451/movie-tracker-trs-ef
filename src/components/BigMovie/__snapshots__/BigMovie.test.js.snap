// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigMovie should render the snapshot 1`] = `
<div
  className="big-movie-card"
>
  <div
    className="movie-poster"
  >
    <img
      alt="poster"
      src="http://image.tmdb.org/t/p/original//2WfjB6FUDTIBVI2y02UGbnHR82s.jpg"
    />
  </div>
  <div
    className="movie-details"
  >
    <h3
      className="movie-title"
    >
      Master Z: Ip Man Legacy
    </h3>
    <h2
      className="release-date"
    >
      Released: 
      2018-12-20
    </h2>
    <h2
      className="vote-avg"
    >
      Rating: 
      5.5
    </h2>
    <p
      className="overview"
    >
      After being defeated by Ip Man, Cheung Tin Chi is attempting to keep a low profile. While going about his business, he gets into a fight with a foreigner by the name of Davidson, who is a big boss behind the bar district. Tin Chi fights hard with Wing Chun and earns respect.
    </p>
  </div>
</div>
`;
